### ISSUE TITLE
Introduce a managed installation of Nix

### ISSUE THREAD

**Issue by williammpratt** [NixOS/nixpkgs#32310]
## Issue description
`curl https://nixos.org/nix/install | sh` is not ready for general usage. 

A user tried to run this script on his MBP and it locked up his machine. After a reboot running it again, results in the user being required to do manual work (looking at /etc/profile for example) with a message that an issue could be opened on GitHub. 

Another usability problem is that no remaining time is displayed. 

This provides damage to the Nix brand, particularly, because there is no inverse operation available. In the words of the user "It created a mess". Whether this is true or not is not relevant. Fact is, that it did cost time and thus money. 

`curl https://nixos.org/nix/uninstall | sh`. 

To summarize, we need two API calls. One to install Nix, one to remove Nix. Reinstalling Nix by calling the install Nix API should always result in installation if a user hasn't modified their file in between the two attempts to install Nix. Any other solution cannot be used as a management solution. 

Just having one script which also takes an uninstall option would also be an option. Not requiring sudo access would be even better, if that's at all possible.

## Technical details
MacOS Sierra 10.13.1

**Comment by FRidh** [comment]
Thanks for the reporting, but this is the wrong issue tracker. Please have a look at https://github.com/NixOS/nix.
